azure-functions-durable/src/lib.rs

```diff
@@ -1,2 +1,310 @@
 //! # Durable Functions for Rust
 #![feature(async_await)]
+
+#[macro_use]
+extern crate derive_builder;
+
+use std::result::Result::*;
+use chrono::prelude::*;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+
+/// Represents the Durable Functions client creation URLs.
+#[derive(Debug, Clone, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CreationUrls {
+    /// The URL for creating a new orchestration instance.
+    #[serde(rename = "createNewInstancePostUri")]
+    pub create_new_instance_url: String,
+    /// The URL for creating and waiting on a new orchestration instance.
+    #[serde(rename = "createAndWaitOnNewInstancePostUri")]
+    pub create_new_instance_and_wait_url: String,
+}
+
+/// Represents the Durable Functions client management URLs.
+#[derive(Debug, Clone, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ManagementUrls {
+    /// The ID of the orchestration instance.
+    pub id: String,
+    /// The status URL of the orchestration instance.
+    #[serde(rename = "statusQueryGetUri")]
+    pub status_query_url: String,
+    /// The "raise event" URL of the orchestration instance.
+    #[serde(rename = "sendEventPostUri")]
+    pub raise_event_url: String,
+    /// The "terminate" URL of the orchestration instance.
+    #[serde(rename = "terminatePostUri")]
+    pub terminate_url: String,
+    /// The "rewind" URL of the orchestration instance.
+    #[serde(rename = "rewindPostUri")]
+    pub rewind_url: String,
+    /// The "purge history" URL of the orchestration instance.
+    #[serde(rename = "purgeHistoryDeleteUri")]
+    pub purge_history_url: String,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct DurableOrchestrationClientBinding {
+    #[serde(rename = "taskHubName")]
+    task_hub: String,
+    creation_urls: CreationUrls,
+    management_urls: ManagementUrls,
+}
+
+impl DurableOrchestrationClientBinding {
+    // Mostly for debugging purposes
+    pub fn new(hub:String, create_new:String, create_and_wait:String, id:String, status_query:String, raise:String, terminate:String, rewind:String, purge:String) -> DurableOrchestrationClientBinding {
```

Comment:
Since we're keeping `CreationUrls` and `ManagementUrls` internally, perhaps we should just take them as arguments to `new`?